optimizely/client/factory.go

```diff
@@ -37,6 +36,7 @@ type Options struct {
 	Context              context.Context
 	ProjectConfigManager optimizely.ProjectConfigManager
 	DecisionService      decision.Service
+	EventProcessor       *event.QueueingEventProcessor
```

Comment:
This is wrong. It should reference the `Processor` interface and not the struct implementation